### PR TITLE
Rename dppl to dppy

### PR DESCRIPTION
**Pull request by 1e-to**
Closes the issue https://github.com/IntelPython/numba-dppy/issues/8

Files in which it was not possible to rename, due to connection with dpCtl:

- numba_dppy/dppy_host_fn_call_gen.py
- numba_dppy/experimental_numpy_lowering_overload.py
- numba_dppy/initialize.py

Locally tested:
    Ran 140 tests in 28.366s
    OK (skipped=26, expected failures=1)

### CODE REVIEW

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Please, update with main branch. It will fix CI.

**Comment by 1e-to** [comment]
I'll tested it with https://github.com/IntelPython/dpctl/pull/201

Ran 140 tests in 28.973s
OK (skipped=26, expected failures=1)

dpctl                     0.3.8+100.gba65df7          pypi_0    pypi

File: numba_dppy/__init__.py [hunk 1]
```diff
@@ -71,13 +71,13 @@ def foo(in, out):
     The following table has the list of intrinsic functions that can be directly
     used inside a DK function. All the functions are equivalent to the similarly
     named OpenCL function. Wherever there is an implementation difference
-    between the Numba-PyDPPL version and the OpenCL version, the difference is
+    between the Numba-PyDPPY version and the OpenCL version, the difference is
```
Comment (diptorupd) [inline numba_dppy/__init__.py:74]: Maybe `Numba-DPPY` instead of `Numba-PyDPPY`

File: numba_dppy/__init__.py [hunk 2]
```diff
@@ -509,11 +509,11 @@ def main():
 if dppy_present:
     from .device_init import *
 else:
-    raise ImportError("Importing dppl failed")
+    raise ImportError("Importing dppy failed")
```
Comment (diptorupd) [inline numba_dppy/__init__.py:512]: `numba-dppy` instead of `dppy`. Our extension package is called `numba-dppy`.

**Review by diptorupd — APPROVED** [review]
Apart from minor changes. LGTM!

**Comment by diptorupd** [comment]
@PokhodenkoSA I have merged master and made couple of changes to let numba-dppy build with dpctl/master. CI needs to be updated to use dpctl master.

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -23,6 +23,7 @@ requirements:
     run:
         - python
         - numba >=0.51
+        - cffi
```
Comment (PokhodenkoSA) [inline conda-recipe/meta.yaml]:
```suggestion
```

File: setup.py [hunk 4]
```diff
@@ -38,6 +38,7 @@ def get_ext_modules():
 build_requires = ["cython"]
 install_requires = [
     "numba",
+    "cffi",
```
Comment (PokhodenkoSA) [inline setup.py]:
```suggestion
```

File: numba_dppy/tests/__init__.py [hunk 5]
```diff
@@ -2,16 +2,17 @@
 from numba.testing import load_testsuite
 from os.path import dirname, join
 
-
+import numba_dppy
 import numba_dppy.config as dppy_config
 
+# from numba_dppy.tests.dppy import *
+
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/__init__.py:9]: Clean commented code.

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
It seems that branch is not up tp date with main branch.

**Review by PokhodenkoSA — APPROVED** [review]

**Pull request merged**